packages/next/build/webpack-config.js

```diff
@@ -294,10 +294,13 @@ export default async function getBaseWebpackConfig (dir, {dev = false, isServer
       new webpack.DefinePlugin(Object.assign(
         {},
         config.env ? Object.keys(config.env)
-          .reduce((acc, key) => ({
-            ...acc,
-            ...{ [`process.env.${key}`]: JSON.stringify(config.env[key]) }
-          }), {}) : {},
+          .reduce((acc, key) => {
+            if (/^(?:NODE_.+)|(?:__.+)$/.test(key.toUpperCase())) throw new Error(`Next.js config env cannot have key of ${key}`)
```

Comment:
Let's add an err.sh link here explaining why the error happens.

Reply:
What's to explain?

Reply:
We want all errors to have an err.sh link eventually. We can explain alternatives and why we block `__` and `NODE_`

Reply:
As it might not be clear for users why we block it.